### PR TITLE
fix: long user name breaks self profile view layout

### PR DESCRIPTION
**Pull request by billypchan**
### Issues

long user name breaks self-profile view layout

### Causes

The label's horizontal `ContentCompressionResistancePriority` is too high.

### Solutions

Lower horizontal `ContentCompressionResistancePriority` to `.defaultLow` and set `lineBreakMode` ` .byTruncatingTail` to limit the width of the label.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/SelfProfile/ProfileView.swift [hunk 1]
```diff
@@ -62,7 +62,7 @@ final class ProfileView: UIView, Themeable {
     /// The view controller that displays the view.
     weak var source: UIViewController?
     
-    @objc dynamic var colorSchemeVariant: ColorSchemeVariant = ColorScheme.default.variant {
+    var colorSchemeVariant: ColorSchemeVariant = ColorScheme.default.variant {
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/SelfProfile/ProfileView.swift]: Why remove the `dynamic` annotation?

File: Wire-iOS/Sources/Components/CustomSpacingStackView.swift [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 import UIKit
 
-@objcMembers class CustomSpacingStackView: UIView {
+final class CustomSpacingStackView: UIView {
```
Comment (typfel) [inline Wire-iOS/Sources/Components/CustomSpacingStackView.swift]: Let's not do unrelated changes.

File: Wire-iOS/Sources/UserInterface/SelfProfile/ProfileView.swift (hunk 1 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/SelfProfile/ProfileView.swift]: It is no longer necessary after its callers are converted to Swift.

Comment (typfel) [reply Wire-iOS/Sources/UserInterface/SelfProfile/ProfileView.swift]: The dynamic annotation is there to support `UIAppearance` which relies on the method being dynamic.
